File: backend/core/domain/engine/Bitboard.py
class Bitboard:
    def __init__(self, *args):
        if len(args) == 1:
            self.bitboard = args[0]
        else:
            self.bitboard = 0

    def add_piece(self, square_no):
        self.bitboard |= 1 << square_no

    def __str__(self):
        return format(self.bitboard, '064b')

    def get_8by8_board(self):
        bit_string = format(self.bitboard, '064b')
        formatted_string = '\n'.join(bit_string[i:i + 8] for i in range(0, 64, 8))
        return formatted_string

    def __add__(self, other):
        return Bitboard(self.bitboard | other.bitboard)

    def __or__(self, other):
        return Bitboard(self.bitboard | other.bitboard)

    def __xor__(self, other):
        return Bitboard(self.bitboard ^ other.bitboard)

    def __and__(self, other):
        return Bitboard(self.bitboard & other.bitboard)

File: backend/ChessEngine.Domain/Models/Bitboard.cs
namespace ChessEngine.Domain.Models
{
    public class Bitboard
    {
        public ulong BitboardValue { get; set; }

        public Bitboard(ulong bitboardValue = 0)
        {
            BitboardValue = bitboardValue;
        }

        public void AddPiece(int squareNo)
        {
            BitboardValue |= 1UL << squareNo;
        }

        public override string ToString()
        {
            return Convert.ToString((long)BitboardValue, 2).PadLeft(64, '0');
        }

        public string Get8By8Board()
        {
            string bitString = Convert.ToString((long)BitboardValue, 2).PadLeft(64, '0');
            string formatted = "";
            for (int i = 0; i < 64; i += 8)
            {
                formatted += bitString.Substring(i, 8) + "\n";
            }
            return formatted;
        }

        public static Bitboard operator +(Bitboard a, Bitboard b)
        {
            return new Bitboard(a.BitboardValue | b.BitboardValue);
        }

        public static Bitboard operator |(Bitboard a, Bitboard b)
        {
            return new Bitboard(a.BitboardValue | b.BitboardValue);
        }

        public static Bitboard operator ^(Bitboard a, Bitboard b)
        {
            return new Bitboard(a.BitboardValue ^ b.BitboardValue);
        }

        public static Bitboard operator &(Bitboard a, Bitboard b)
        {
            return new Bitboard(a.BitboardValue & b.BitboardValue);
        }
    }
}